### PR TITLE
WIP Remove uses of ipython_genutils.py3compat  in favor of six

### PR DESCRIPTION
**Pull request by SylvainCorlay**

### CODE REVIEW

**Comment by jdfreder** [comment]
+1 :star:

File: traitlets/config/application.py [hunk 1]
```diff
@@ -27,7 +27,9 @@
 from ipython_genutils.importstring import import_item
 from ipython_genutils.text import indent, wrap_paragraphs, dedent
 from ipython_genutils import py3compat
-from ipython_genutils.py3compat import string_types, iteritems
+
+import six
+from six import iteritems
```
Comment (minrk) [inline traitlets/config/application.py]: I would actually just get rid of all of our uses of iteritems. I don't think we ever use it in a case where it's beneficial.

Comment (SylvainCorlay) [inline traitlets/config/application.py]: so just `.items`?

Comment (minrk) [inline traitlets/config/application.py]: Yup. We aren't operating on any giant dicts here. We can afford to let Python 2 be the tiniest bit slower on what we are dealing with. No need to call in six for this one.

**Comment by Carreau** [comment]
Hum, this will need rebase.

**Comment by SylvainCorlay** [comment]
Yeah it is stale, I have to get back to this one.

**Pull request closed without merging**